### PR TITLE
Fetch single image result in `asyncData` hook instead of `fetch`

### LINKED ISSUE #980: Navigate to the image detail page with an invalid id breaks it
## Description

The image details page does not handle invalid or null `id` well.

## Reproduction

1. Go to https://search-staging.openverse.engineering/image/foo
2. See a blank page.

or alternately:

1. Run the project and go to http://localhost:8443/image/foo
2. See the error page but inspect the browser console and the terminal.

## Expectation

When the page gets an invalid or null `id` it should show the error page and return a 404 response status code.

### On staging
CleanShot 2022-02-21 at 14 53 00@2x

### Locally
CleanShot 2022-02-21 at 14 52 35@2x

## Additional context

Seems to be a residue or a more general case of #496.

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Fixes #980 by @krysal

## Description

This page was using the `fetch` hook when it should be [`asyncData`](https://nuxtjs.org/docs/features/data-fetching#async-data) hook of Nuxt as in the audio page, the last one is the correct for fetching async data on the server-side.

Managing invalid or null ids this way we no longer need the `index.vue` pages for /image and /audio, and these were also returning inaccurate 200 status codes with the layout for a 404 error page.

## Testing Instructions

Reproduce steps in the mentioned issue and observe the expected behavior: the error is handled correctly showing the error page without fetching and a 404 status code is returned.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/pages/audio/index.vue [hunk 1]
```diff
@@ -1,11 +0,0 @@
...
-
-<script>
-import { defineComponent } from '@vue/composition-api'
-import VFourOhFour from '~/components/VFourOhFour.vue'
-
-export default defineComponent({
-  components: { VFourOhFour },
-  layout: 'blank',
-})
-</script>
```
Comment (sarayourfriend) [inline src/pages/audio/index.vue:11]: I _knew_ there had to be a better way to do this. Thank you for removing the need for these!

**Review by sarayourfriend — APPROVED** [review]
LGTM! Verified that this fixes the issue compared to staging.

Seriously so glad you were able to get rid of those two `index.vue` pages :rocket:

**Comment by obulat** [comment]
This is a very good solution to the 404 problem.
If I remember correctly, it was done as this (with fetch and thumbnail) to make the navigation seamless: you would see the thumbnail at first, and then, after the actual image loads, you would see (but probably not notice the change to) the actual image.
Now, when you navigate to the detail page, you (or at least I) can see the blank image section, and then the image pops up into the view. 
@zackkrida , was my explanation correct?

**Comment by sarayourfriend** [comment]
>  If I remember correctly, it was done as this (with fetch and thumbnail) to make the navigation seamless: you would see the thumbnail at first, and then, after the actual image loads, you would see (but probably not notice the change to) the actual image.
> Now, when you navigate to the detail page, you (or at least I) can see the blank image section, and then the image pops up into the view.

This was my understanding of the _intended_ behavior but honestly I haven't seen that working in months.

If I go to https://search-staging.openverse.engineering/search/?q=galah and click any of the image results, I see the same behavior you described, Olga, of the space for the image being blank and then popping in afterwards.

**Comment by krysal** [comment]
@obulat that makes sense, I can see that behavior in production. Fortunately, that's easy to add! :)

**Review by zackkrida — APPROVED** [review]
@krysal ideally, we would only show the thumbnail first _if_ the user is navigating from the search result page, where the thumbnail has already been loaded. With the current approach, any user visiting a server-rendered image url directly, ends up downloading two images, the thumbnail and the full image, when it would be better to only download the full image.

I think this is okay to merge without fixing this, since the current solution optimizes for the most common use case, but it would be good to fix in a separate issue.

**Comment by krysal** [comment]
@zackkrida That optimization sounds possible and good for a following "PR", here I was thinking of just keeping the old behavior after fixing the bug. Thanks for reviewing!

**Pull request merged**